src/Swarm/Game/Step.hs

```diff
@@ -748,6 +749,46 @@ stepCESK cesk = case cesk of
   ------------------------------------------------------------
   -- Execution
 
+  -- Executing a 'requirements' command generates an appropriate log message
+  -- listing the requirements of the given expression.
+  Out (VRequirements src t _) s (FExec : k) -> do
+    currentContext <- use $ robotContext . defReqs
+    em <- use entityMap
+    let (R.Requirements caps devs inv, _) = R.requirements currentContext t
+        deviceSets :: Set (Set Text)
+        deviceSets =
+          -- Remove any device sets which are a superset of another
+          -- set.  For example, if (grabber OR fast grabber OR
+          -- harvester) is required but (grabber OR fast grabber) is
+          -- also required then we might as well remove the first set,
+          -- since satisfying the second device set will automatically
+          -- satisfy the first.
```

Comment:
Are both sets equally likely to be the one removed? Or is there bias toward removing the larger set?

Reply:
A set will be removed if and only if it is a superset of another set.  So the larger set will always be removed.